hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/AsyncCompactService.java

```diff
@@ -78,13 +81,26 @@ protected Pair<CompletableFuture, ExecutorService> startService() {
 
         while (!isShutdownRequested()) {
           final HoodieInstant instant = fetchNextAsyncServiceInstant();
-
           if (null != instant) {
             LOG.info("Starting Compaction for instant " + instant);
+            synchronized (writeConfigUpdateLock) {
```

Comment:
from my understanding of previous comments, this still allows async service to receive write config updates? if we think about async service coupling with a table service client and being immutable, we shouldn't need this sort of lock. we just create new async service at deltasync level, and freeze write config for the service internally.

Reply:
we can't cancel existing async service if its inflight. And so we might have to create N such instances and need to find a way to clean those up. Wouldn't this be easier than that.